R/zchunk_LB162.ag_prodchange_R_C_Y_GLU_irr.R

```diff
@@ -41,67 +45,460 @@ module_aglu_LB162.ag_prodchange_R_C_Y_GLU_irr_DISABLED <- function(command, ...)
+                by = c("CROSIT_ctry", "CROSIT_crop", "Irr_Rfd")) %>%
+      # repeat for all years in SPEC_AG_PROD_YEARS and join Yield multipliers for each year
+      repeat_add_columns(tibble::tibble(year = SPEC_AG_PROD_YEARS)) %>%
+      left_join_error_no_match(L162.ag_Yieldmult_Rcrs_Ccrs_Y_irr,
+                               by = c("CROSIT_ctry", "CROSIT_crop", "Irr_Rfd", "year")) ->
+      L162.ag_HA_ha_Rcrs_Ccrs_Ysy_GLU_irr
+
+
+
+    # Calculating the adjusted production / harvested area in each time period, for each GCAM region / commodity / GLU.
+    # Starting from full GTAP table for composite regions and commodities in the CROSIT database, L162.ag_HA_ha_ctry_crop_irr,
+    # repeating for all years in SPEC_AG_PROD_YEARS.
+    # Then join in yield multipliers from CROSIT L162.ag_HA_ha_Rcrs_Ccrs_Ysy_GLU_irr, GCAM region and GCAM commodity information.
+    # Calculate modified production, Prod_mod = base year harvested area HA * yearly yield multipliers Mult. Production and
+    # Harvested area are then aggregated to the GCAM region - commodity level so that Aggregated Yield can be correctly calculated.
+    # The YieldRatio = Prod_mod/HA is then calculated and output for each GCAM region-commodity-glu-irrigation-year.
+    #
+    # There is a small error in the old data system. The intention is to match in multipliers by CROSIT_ctry, CROSIT_crop, irrigation,
+    # and year. The old DS only does so by CROSIT_ctry, CROSIT_crop, and year.
+
+    if(OLD_DATA_SYSTEM_BEHAVIOR) {
+      # preprocess table of multipliers before joining, restricting to the CROSIT country-crop-glu-irrigation present in
+      # the full GTAP table,  L162.ag_HA_ha_ctry_crop_irr
+      L162.ag_HA_ha_Rcrs_Ccrs_Ysy_GLU_irr %>%
+        select(CROSIT_ctry, CROSIT_crop, year, Mult) %>%
+        semi_join(L162.ag_HA_ha_ctry_crop_irr, by = c("CROSIT_ctry", "CROSIT_crop")) %>%
+        dplyr::distinct() ->
+        CROSIT_mult
+
+      L162.ag_HA_ha_ctry_crop_irr %>%
+        na.omit() %>%
+        repeat_add_columns(tibble::tibble(year = SPEC_AG_PROD_YEARS)) %>% # agree fine to here; it's the next join
+        left_join_keep_first_only(CROSIT_mult, by = c("CROSIT_ctry", "CROSIT_crop", "year")) %>%
+        na.omit() %>%
+        left_join_error_no_match(select(iso_GCAM_regID, iso, GCAM_region_ID), by = "iso") %>%
+        left_join_error_no_match(select(FAO_ag_items_PRODSTAT, GTAP_crop, GCAM_commodity), by = "GTAP_crop") %>%
+        # Multiply base-year harvested area by the future productivity multipliers to calculate prod_mod and aggregate
+        mutate(Prod_mod = HA * Mult)  %>%
+        group_by(GCAM_region_ID, GCAM_commodity, year, GLU, Irr_Rfd) %>%
+        summarise(HA = sum(HA), Prod_mod = sum(Prod_mod)) %>%
+        ungroup() %>%
+        # Calculate YieldRatio = Prod_mod/HA by region-commodity-glu-irrigation-year; subset and output the YieldRatios
+        mutate(YieldRatio = Prod_mod / HA) %>%
```

Comment:
I used YieldRatio whenever the value was the yield the given year divided by the yield in the base year. YieldRate is the annual growth rate applied in the given model time step. The YieldRatio is more versatile because its value is not influenced by the choice of model future years, but the YieldRate is the parameter that is ultimately read in to the model.